pkg/graveler/graveler_test.go

```diff
@@ -2292,3 +2292,182 @@ func TestGravelerCreateCommitRecord(t *testing.T) {
 		require.NoError(t, err)
 	})
 }
+
+func TestGraveler_Revert(t *testing.T) {
+	type fields struct {
+		CommittedManager *testutil.CommittedFake
+		RefManager       *testutil.RefsFake
+	}
+	type args struct {
+		ctx          context.Context
+		branchID     graveler.BranchID
+		ref          graveler.Ref
+		parentNumber int
+		allowEmpty   bool
+	}
+	tests := []struct {
+		name        string
+		fields      fields
+		args        args
+		expectedVal graveler.CommitID
+		expectedErr error
+	}{
+		{
+			name: "ref not found",
+			fields: fields{
+				CommittedManager: &testutil.CommittedFake{},
+				RefManager: &testutil.RefsFake{
+					Branch: &graveler.Branch{CommitID: "c1"},
+					Commits: map[graveler.CommitID]*graveler.Commit{
+						"c1": {MetaRangeID: "mri1"},
+					},
+					Refs: map[graveler.Ref]*graveler.ResolvedRef{
+						"b1": {
+							Type:                   graveler.ReferenceTypeBranch,
+							ResolvedBranchModifier: 0,
+							BranchRecord: graveler.BranchRecord{
+								BranchID: "b1",
+								Branch: &graveler.Branch{
+									CommitID: "c1",
+								},
+							},
+						},
+						"c1": {
+							Type:                   graveler.ReferenceTypeCommit,
+							ResolvedBranchModifier: 0,
+							BranchRecord: graveler.BranchRecord{
+								Branch: &graveler.Branch{
+									CommitID: "c1",
+								},
+							},
+						},
+					},
+				},
+			},
+			args: args{
+				ctx:          nil,
+				branchID:     "b1",
+				ref:          graveler.Ref("ref1"),
+				parentNumber: 0,
+				allowEmpty:   false,
+			},
+			expectedErr: graveler.ErrCommitNotFound,
+			expectedVal: "",
+		},
+		{
+			name: "fail on staging token",
+			fields: fields{
+				CommittedManager: &testutil.CommittedFake{},
+				RefManager: &testutil.RefsFake{
+					Branch: &graveler.Branch{CommitID: "c1", StagingToken: "token"},
+					Commits: map[graveler.CommitID]*graveler.Commit{
+						"c1": {MetaRangeID: "mri1"},
+					},
+					Refs: map[graveler.Ref]*graveler.ResolvedRef{
+						"b1": {
+							Type:                   graveler.ReferenceTypeBranch,
+							ResolvedBranchModifier: 0,
+							BranchRecord: graveler.BranchRecord{
+								BranchID: "b1",
+								Branch: &graveler.Branch{
+									CommitID:     "c1",
+									StagingToken: "token",
```

Comment:
Followup on comment above about changing object names in the test cases: Here I might prefer
```suggestion
							BranchRecord: graveler.BranchRecord{
								BranchID: "dirty-branch",
								Branch: &graveler.Branch{
									CommitID:     "c1",
									StagingToken: "token",
```